### PR TITLE
Prevent asserting with significant expressions

### PR DESCRIPTION
**Pull request by legendecas**
Also renamed category `node-api` to `napi`

Additional to #79.

### CODE REVIEW

File: async_work_thread_safe_function/napi/binding.c [hunk 1]
```diff
@@ -85,26 +91,31 @@ static void ExecuteWork(napi_env env, void* data) {
...
-                                          napi_tsfn_release) == napi_ok);
+  status = napi_release_threadsafe_function(addon_data->tsfn,
+                                          napi_tsfn_release);
+  assert(status == napi_ok);
 }
 
 // This function runs on the main thread after `ExecuteWork` exits.
 static void WorkComplete(napi_env env, napi_status status, void* data) {
   AddonData* addon_data = (AddonData*)data;
+  napi_status s;
```
Comment (mhdawson) [inline async_work_thread_safe_function/napi/binding.c]: A nit but I'd prefer a more complete name, for example `status` versus the one letter name

**Comment by legendecas** [comment]
@mhdawson updated :)

**Comment by legendecas** [comment]
@mhdawson this PR contains another commit from https://github.com/nodejs/node-addon-examples/pull/79 that I think it's worth to maintain the original authorship. WDYT? Maybe a force-push is needed.

**Comment by mhdawson** [comment]
@legendecas most often we squash all of the comments which is what I did in this case.  I see there is a PR from another commit. We probably should have landed that separately by fixing up the conflicts in the PR if possible.

**Pull request merged**